packages/next/build/webpack/plugins/next-esm-plugin.ts

```diff
@@ -131,21 +132,37 @@ export default class NextEsmPlugin implements Plugin {
 
     const childChunkFileMap = childCompilation.chunks.reduce(
       (chunkMap: { [key: string]: any[] }, chunk: compilation.Chunk) => {
-        chunkMap[chunk.name] = chunk.files
+        chunkMap[chunk.name] = chunk
         return chunkMap
       },
       {}
     )
 
+    // Merge files from similar chunks
     compilation.chunks.forEach((chunk: compilation.Chunk) => {
-      const childChunkFiles = childChunkFileMap[chunk.name]
+      const childChunk = childChunkFileMap[chunk.name]
 
-      if (childChunkFiles) {
+      if (childChunk && childChunk.files) {
+        delete childChunkFileMap[chunk.name]
         chunk.files.push(
-          ...childChunkFiles.filter((v: any) => !chunk.files.includes(v))
+          ...childChunk.files.filter((v: any) => !chunk.files.includes(v))
         )
       }
     })
+
+    // Add modern only chunks
+    compilation.chunks.push(...Object.values(childChunkFileMap))
```

Comment:
The modern build will have different chunks with the new config. This code makes sure they are added inside the right entry point